library/debug.c

```diff
@@ -227,6 +227,8 @@ void mbedtls_debug_print_mpi( const mbedtls_ssl_context *ssl, int level,
         NULL == ssl->conf        ||
         NULL == ssl->conf->f_dbg ||
         NULL == X                ||
+        NULL == X->p             ||
+        0    == X->n             ||
```

Comment:
Stopping here if `X->n == 0` is wrong. That's zero, and it's a perfectly cromulent number. It just needs to be printed correctly.